engine.py

```diff
@@ -327,105 +327,134 @@ def register_command(self, name, callback, properties=None):
             properties,
         )
 
-    def generate_thumbnail(self, document=None, output_path=None):
+    def export_as_jpeg(self, document=None, output_path=None, max_size=2048, quality=12):
         """
-        Try to generate a thumbnail for an open document.
-
-        If a thumbnail can be generated, the output path will be returned. If
-        no thumbnail can be created, ``None`` will be returned.
-
+        Export a Jpeg image from the given document or from the current document.
+        
         :param document: The document to generate a thumbnail for. Assumes the
-            active document if ``None`` is supplied.
+                         active document if ``None`` is supplied.
         :param output_path: The output file path to write the thumbnail. If
-            ``None`` supplied, the method will write to a temp file.
-        :return:
+                            ``None`` is supplied, the method will write to a temp file.
+        :param int max_size: The maximum width and height of the exported image.
+        :param int quality: The Jpeg quality of the exported image.
+        :returns: The full path to the exported image.
+        :raises: RuntimeError if the document or its size can't be retrieved.
         """
+        adobe = self.adobe
 
-        # set unit system to pixels:
-        original_ruler_units = self.adobe.app.preferences.rulerUnits
-        pixel_units = self.adobe.Units.PIXELS
+        # Get some current values so we can restore them.
+        original_ruler_units = adobe.app.preferences.rulerUnits
+        original_dialog_mode = adobe.app.displayDialogs
 
-        self.adobe.app.preferences.rulerUnits = pixel_units
+        # If no output_path was given, use a temp file.
+        jpeg_pub_path = output_path or os.path.join(
+            tempfile.gettempdir(), "%s_sgtk.jpg" % uuid.uuid4().hex
+        )
 
         with self.context_changes_disabled():
-
-            # we need to set the units back no matter what
             try:
-
-                # make sure we have a document
-                if not document:
-                    try:
-                        document = self.adobe.app.activeDocument
-                    except RuntimeError:
-                        raise Exception("There is no active document!")
-
-                orig_name = document.name
-                width_str = str(document.width)
-                height_str = str(document.height)
-
-                # build temp name for the thumbnail doc (just in case we fail to
-                # close it!):
+                # Set unit system to pixels:
+                adobe.app.preferences.rulerUnits = adobe.Units.PIXELS
+                # Disable dialogs.
+                adobe.app.displayDialogs = adobe.DialogModes.NO
+
+                active_doc = document or adobe.app.activeDocument
```

Comment:
This will raise if there's no active document, which is why Josh had it wrapped in a try/except. I thought we had a method on AdobeBridge to get this, but it looks like it specifically returns the active document path on disk and not the document object. I think it's worth adding a get_active_document method there and have it handle the possible RuntimeError and return None if that happens.

The reason I say wrap it is because the error message that comes back from the RPC API is going to be fairly useless, because Photoshop's CEP failure messages are likewise worthless to a user.